### DISCUSSION TITLE
TGA error no wall or particle to which to apply the TGA analysis

### DISCUSSION THREAD

**Discussion by bjmvandongen** [firemodels/fds#13211] [Q&A]
Hi all,

For a case where I am trying to mimic experiments from Mendes Lopes et al, I want to do a TG analysis in FDS. 
But when I do this, I get the error:  No wall or particle to which to apply the TGA analysis. I don't understand this because I placed       TGA_ANALYSIS         = T on the SURF line of my vegetation particles. 
I added my code below.

Does anybody know what is wrong with the code or why I get the error?
[TGA_Mendes.txt](https://github.com/user-attachments/files/16359309/TGA_Mendes.txt)

**Comment by drjfloyd** [comment]
TGA_ANALYSIS doesn't play well with multiple meshes. Make a simple input file with just a small MESH, your MATL and SURF you want to analyze, and a VENT or PART in that small MESH with the SURF_ID.   

There is a caution statement in the User's Guide (see below). It should, however, be a much stronger statement, i.e. only use a single mesh.  

Note: If applying TGA_ANALYSIS to a multi-mesh case or a case where the SURF is associated with a PART
class yet to be inserted into the calculation, FDS may not find the SURF and hence throw an error. It is best
to simplify the domain to a single particle or wall cell in a single mesh.

Make a simpler input with a single mesh.

**Reply by bjmvandongen** [reply]
Thank you very much for the comment. I will make my case much more simple with one mesh.

**Comment by mcgratta** [comment]
I'll make TGA play nice with multiple meshes.

**Comment by mcgratta** [comment]
TGA works now for multiple mesh cases. To test it, download and install a nightly build.
